src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/GATKDefaultCLPConfigurationArgumentCollection.java

```diff
@@ -0,0 +1,102 @@
+    public static final String TMP_DIR_NAME = "TMP_DIR";
+    public static final String QUIET_NAME = "QUIET";
+    public static final String USE_JDK_DEFLATER_LONG_NAME = "use-jdk-deflater";
+    public static final String USE_JDK_DEFLATER_SHORT_NAME = "jdk-deflater";
+    public static final String USE_JDK_INFLATER_LONG_NAME = "use-jdk-inflater";
+    public static final String USE_JDK_INFLATER_SHORT_NAME = "jdk-inflater";
+    public static final String NIO_MAX_REOPENS_LONG_NAME = "gcs-max-retries";
+    public static final String NIO_MAX_REOPENS_SHORT_NAME = "gcs-retries";
+
+    @Argument(fullName = TMP_DIR_NAME, doc = "List of temp directories to use.", common=true, optional=true)
+    public List<String> tmpDir = new ArrayList<>();
+
+    @Argument(fullName = StandardArgumentDefinitions.VERBOSITY_NAME, shortName = StandardArgumentDefinitions.VERBOSITY_NAME, doc = "Control verbosity of logging.", common = true, optional = true)
+    public Log.LogLevel verbosity = Log.LogLevel.INFO;
+
+    @Argument(fullName = QUIET_NAME, doc = "Whether to suppress job-summary info on System.err.", common=true)
+    public Boolean quiet = false;
+
+    @Argument(fullName = USE_JDK_DEFLATER_LONG_NAME, shortName = USE_JDK_DEFLATER_SHORT_NAME, doc = "Whether to use the JdkDeflater (as opposed to IntelDeflater)", common=true)
+    public boolean useJdkDeflater = false;
+
+    @Argument(fullName = USE_JDK_INFLATER_LONG_NAME, shortName = USE_JDK_INFLATER_SHORT_NAME, doc = "Whether to use the JdkInflater (as opposed to IntelInflater)", common=true)
+    public boolean useJdkInflater = false;
+
+    @Argument(fullName = NIO_MAX_REOPENS_LONG_NAME, shortName = NIO_MAX_REOPENS_SHORT_NAME, doc = "If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection", optional = true)
+    public int nioMaxReopens = ConfigFactory.getInstance().getGATKConfig().gcsMaxRetries();
+
+    // This option is here for documentation completeness.
+    // This is actually parsed out in Main to initialize configuration files because
+    // we need to have the configuration completely set up before we create our CommandLinePrograms.
+    // (Some of the CommandLinePrograms have default values set to config values, and these are loaded
+    // at class load time as static initializers).
+    @Argument(fullName = StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION,
+            doc = "A configuration file to use with the GATK.",
+            common = true,
+            optional = true)
+    public String gatkConfigFile = null;
+
+    private final List<Path> resolvedTmpDirectories = new ArrayList<>();
+
+    @Override
+    public List<Path> getTmpDirectories() {
+        if (resolvedTmpDirectories.isEmpty()) {
+            // Provide one temp directory if the caller didn't
+            if (tmpDir == null || tmpDir.isEmpty()) {
+                resolvedTmpDirectories.add(IOUtil.getDefaultTmpDirPath());
```

Comment:
In addition, it looks like the `IOUtil.getDefaultTmpDirPath` is broken in HTSJDK, because it keeps generating in sub-subsequent calls the `urer.dir`: in my local computer, there are some tests where there are at least 50 `daniel` sub-folders!